### PR TITLE
[SPARK-49494][DOCS] Use spark logos from spark-website in docs

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

Use spark logos from spark-website in docs and remove them from spark main repo

### Why are the changes needed?

- reduce spark-website incremental size
- fix some cases when `/img/spark-logo-rev.svg` is invalid

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
doc build, https://github.com/yaooqinn/spark/actions/runs/10676269598/artifacts/1884295142

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @dongjoon-hyun @cloud-fan @HyukjinKwon, thank you

**Review by dongjoon-hyun — COMMENTED** [review]
I'm wondering this is a safe way which is able to provide a recovery from the disaster, @yaooqinn . 

For example, how can we recover our website when ASF Infra team lost the files at https://spark.apache.org/images/spark-logo-rev.svg ?

**Comment by yaooqinn** [comment]
Thank you @dongjoon-hyun. I'm not using official logos in the infrastructure service of https://apache.org/logos since we have not fully uploaded all our logo variants to it.

The https://spark.apache.org/images/spark-logo-rev.svg points to https://github.com/apache/spark-website/blob/asf-site/site/images/spark-logo-rev.svg which is a copy from https://github.com/apache/spark-website/blob/asf-site/images/spark-logo-rev.svg by jekyll build. I guess we handle these files ourselves.

In this case, if the `spark-logo-rev.svg` is 'lost', which probably means that the entire Apache Web Service is in downtime, and https://spark.apache.org is also unreachable.

**Comment by dongjoon-hyun** [comment]
Maybe, I wasn't clear. What I meant is the case ASF website lost the disk and we need to republish our website cleanly. Let me rephrase my question more specifically. So, all deleted files of this PR are redundant ones because our `spark-website.git` already has them?

> In this case, if the spark-logo-rev.svg is 'lost', which probably means that the entire Apache Web Service is in downtime, and https://spark.apache.org/ is also unreachable.

**Comment by yaooqinn** [comment]
Yes, @dongjoon-hyun. The 'spark-website.git' has them

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Merged to master! Thank you.
